back/engines/free/email_campaigns/app/services/email_campaigns/side_fx_campaign_service.rb

```diff
@@ -9,12 +9,16 @@ def before_create(campaign, user); end
     def before_update(campaign, user); end
 
     def after_update(campaign, _user)
-      # We currently only have one campaign that can be enabled/disabled at phase level,
-      # and we align the respective phase-level values with the campaign enabled when the campaign value is updated.
-      return unless campaign.saved_change_to_enabled? &&
-                    campaign.instance_of?(EmailCampaigns::Campaigns::ProjectPhaseStarted)
-
-      toggle_project_phase_started(campaign)
+      # This is a special case for the project_phase_started campaign, the only
+      # campaign that's currently configurable on a phase level.
+      # We turn off all phase email toggles for this campaign when the platform-wide
+      # setting is turned off.
+      # attribute_before_last_save:
+      # https://apidock.com/rails/v6.0.0/ActiveRecord/AttributeMethods/Dirty/attribute_before_last_save
+      if campaign.enabled_before_last_save &&
```

Comment:
It's been a busy day, so I could only review the backend. I think we should add `&& !campaign.enabled` to make sure that the value has changed. Otherwise, I think we might open up the possibility for painful bugs if at some point we would allow also changing other attributes of a campaign.

Reply:
Ok, let's discuss as well during our next 10% session.

Reply:
@sebastienhoorens: Isn't just `!campaign.enabled` instead of `campaign.enabled_before_last_save` enough then?

Reply:
I think we should check both:
1. The campaign was enabled before
2. The campaign is now disabled

Only when both are true we have a true "toggle off" event.

Reply:
Alternatively we could have separate toggle_off/toggle_on endpoints, which feels more robust to me, but that would be more effort.

Reply:
Ok, I have a follow-up question but I'll keep it for our session. Thanks